pkg/storage/factory.go

```diff
@@ -575,8 +577,18 @@ func (c *ClientMetrics) Unregister() {
 	c.AzureMetrics.Unregister()
 }
 
-// NewObjectClient makes a new StorageClient of the desired types.
+// NewObjectClient makes a new StorageClient with the prefix in the front.
 func NewObjectClient(name string, cfg Config, clientMetrics ClientMetrics) (client.ObjectClient, error) {
+	actual, err := internalNewObjectClient(name, cfg, clientMetrics)
+	if err != nil {
+		return nil, err
+	}
+
+	return client.NewPrefixedObjectClient(actual, cfg.ObjectPrefix), nil
```

Comment:
I don't think we should always wrap the object client with the Prefixed object client.

The prefixed object client does string concatenation on almost every API request, this will incur more allocations and a performance penalty that I don't think we should apply in all cases. 

(Note: I'm not sure that allocation an empty string to a string isn't optimized away? so I guess if we could prove that in the empty string case there isn't any penalty we could leave this as is)

Reply:
Done, thank you for the suggestion!